### DISCUSSION TITLE
Charge automation not working

### DISCUSSION THREAD

**Discussion by chowdarygm** [alandtse/tesla#903] [Q&A] [answered]
Newly finished setting up my HA and Tesla integration and all the data seems to be flowing correctly. I did not change the polling interval from default one. I am moving to a hourly pricing through my electricity provider and preparing to set automations for charging when the hourly prices hit certain numbers. In preparation to that I created a simple time based automation to start charging at a particular time for test run but the automation is not running as expected. I did more reading and understand it could be due to the fact that the car goes to sleep and the automation may not be able to wake up the car and send the necessary command. I saw a  script to send a scheduled charging command with wakeup signal. Can I similar script be created for flipping the charging switch with wake up signal when the prices are below certain number and charger is connected? If so, any help with the script is appreciated as i am not good with programming. Thank you.

**Comment by skipishere** [comment]
The turning on of the charger counts as a command that will wake the car up. I have an automation for my Model 3 that does this exact thing (turns the charger on at 0200 until 0500 each morning as that's when my electricity is cheapest).

It is worth checking if you can get the car to accept commands in general as the unofficial API that was being used for some people could read data but not update (and there have been some fixes this week to use the official one)

**Comment by chowdarygm** [comment]
> The turning on of the charger counts as a command that will wake the car up. I have an automation for my Model 3 that does this exact thing (turns the charger on at 0200 until 0500 each morning as that's when my electricity is cheapest).
> 
> It is worth checking if you can get the car to accept commands in general as the unofficial API that was being used for some people could read data but not update (and there have been some fixes this week to use the official one)

Yes, looks like my commands are not working. I tried few commands like lock/unlock and I keep getting below errors (Failed to call service climate/set_hvac_mode. unknown error, Failed to call service lock/unlock. unknown error). I have also attached logs if someone can help with whats happening here?

```
2024-03-14 08:51:59.318 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [140239532360384] 
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 239, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2319, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2356, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 905, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 975, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/lock/__init__.py", line 228, in async_handle_unlock_service
    await self.async_unlock(**self.add_default_code(kwargs))
  File "/config/custom_components/tesla_custom/lock.py", line 43, in async_unlock
    await self._car.unlock()
  File "/usr/local/lib/python3.12/site-packages/teslajsonpy/car.py", line 1177, in unlock
    data = await self._send_command("UNLOCK")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/teslajsonpy/car.py", line 775, in _send_command
    raise ex
  File "/usr/local/lib/python3.12/site-packages/teslajsonpy/car.py", line 760, in _send_command
    data = await self._controller.api(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/teslajsonpy/controller.py", line 1367, in api
    response = await self.__post_with_retries(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/tenacity/_asyncio.py", line 88, in async_wrapped
    return await fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/tenacity/_asyncio.py", line 47, in __call__
    do = self.iter(retry_state=retry_state)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/tenacity/__init__.py", line 325, in iter
    raise retry_exc.reraise()
          ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/tenacity/__init__.py", line 158, in reraise
    raise self.last_attempt.result()
          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/usr/local/lib/python3.12/site-packages/tenacity/_asyncio.py", line 50, in __call__
    result = await fn(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/teslajsonpy/controller.py", line 1390, in __post_with_retries
    return await self.__connection.post(command, method=method, data=data, url=url)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/teslajsonpy/connection.py", line 166, in post
    return await self.__open(url, method=method, headers=self.head, data=data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/teslajsonpy/connection.py", line 218, in __open
    raise TeslaException(resp.status_code)
teslajsonpy.exceptions.TeslaException

```

**Reply by skipishere** [reply, marked as answer]
You need to use the latest version (v3.20.3+) and the new setup process as, by your description, your car won't allow usage of the old unofficial API

**Reply by Uersu8** [reply]
@skipishere could you please add a link to where the "new setup process" is described?

**Reply by skipishere** [reply]
It's in the main [readme](https://github.com/alandtse/tesla?tab=readme-ov-file#tesla-fleet-api-proxy)

**Comment by chowdarygm** [comment]
> You need to use the latest version (v3.20.3+) and the new setup process as, by your description, your car won't allow usage of the old unofficial API

Got it, when the old API method was able to read data, I felt happy as I understand the proxy thing is a complex setup. If I won't be able to run commands with old API, then I will give the new setup process a try.

**Reply by Uersu8** [reply]
Where do I find the "new setup process" description? Could you please add a link?

I am in the same situation that I can read the data from the car, but get the error message if I try to change any setting.

**Comment by AndrewTapp** [comment]
I'm currently using v3.19.11 of this integration with automations to control 'Backup reserve', 'Energy exports' and 'Operation mode' and have done for a number of years.  Despite the issues facing a number of people this integration, at least for me at the moment is working without issue.  My intention is to remain on this version, at least until it breaks!!

> Core 2024.5.2
> Supervisor 2024.05.1
> Operating System 12.3
> Frontend 20240501.1

**Reply by skipishere** [reply]
The issue is more around Tesla deprecating the old way we communicated with the devices (Powerwall/Car) so it's totally at their mercy. The integration still works (for me) using the older API but at some point I'll have to use the Fleet API (which currently is causing more issues due to rate limiting being enabled in the last few days)

**Reply by Boogmeister** [reply]
> The issue is more around Tesla deprecating the old way we communicated with the devices (Powerwall/Car) so it's totally at their mercy. The integration still works (for me) using the older API but at some point I'll have to use the Fleet API (which currently is causing more issues due to rate limiting being enabled in the last few days)

Rate limiting!

DAMMIT!

That's going to be my issue...

Just give me local access FFS Tesla!
